### PR TITLE
Track request duration in changesets.

### PR DESCRIPTION
**Pull request by simozhan**
JIRA: PDC-740

### CODE REVIEW

**Comment by lubomir** [comment]
Looks good to me.

File: pdc/apps/changeset/views.py [hunk 1]
```diff
@@ -80,7 +80,9 @@ def list(self, request, *args, **kwargs):
                     {
                         {
                             "author": "xxx",
+                            "requested_on": "2015-02-03T05:50:49.387Z",
                             "committed_on": "2015-02-03T05:51:17.262Z",
+                            "duration": "27.875",
```
Comment (erichuanggit) [inline pdc/apps/changeset/views.py]: Is it better to append `unit`?

Comment (lubomir) [inline pdc/apps/changeset/views.py]: I don't think so. At most the unit could be mentioned in documentation, but putting it in the response does not seem very useful.

**Comment by erichuanggit** [comment]
yes, what i want to mentioned is adding to doc.

**Comment by lubomir** [comment]
:+1:

File: pdc/apps/changeset/migrations/0004_changeset_requested_on.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    dependencies = [
+        ('changeset', '0003_changeset_pdc_change_comment'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='changeset',
+            name='requested_on',
+            field=models.DateTimeField(default=datetime.datetime(2015, 9, 8, 2, 15, 31, 582918, tzinfo=utc)),
```
Comment (xychu) [inline pdc/apps/changeset/migrations/0004_changeset_requested_on.py]: Will this set existed records 'requested_on' to '2015-9-8'?
What will happen if theirs 'committed_on' are before that value?
I think 'requested_on' should allows to be Null and we could set 'duration' to NaN for historical data that without 'requested_on'.

Comment (lubomir) [inline pdc/apps/changeset/migrations/0004_changeset_requested_on.py]: Another option is to set `requested_on` to the same value as `commited_on`, and old data would show as taking 0 seconds.

Comment (xychu) [inline pdc/apps/changeset/migrations/0004_changeset_requested_on.py]: Much simpler and neater. :+1:

**Comment by ycheng-aa** [comment]
Looks good. And I agree with @erichuanggit , add time unit to doc may be more clear.

File: pdc/apps/changeset/migrations/0004_changeset_requested_on.py (hunk 2 above)
Comment (simozhan) [inline pdc/apps/changeset/migrations/0004_changeset_requested_on.py]: ok. will fix it

**Comment by xychu** [comment]
I think @lubomir means we could add a data migration that set the historical changesets' requested_on to the value of theirs committed_on. So it's more like:
* a migration to add field that allow null;
* a data migration to update historical data;
* another migration to update field that no allow null;

**Comment by lubomir** [comment]
Yes, @xychu is right. Doing it with these migrations means there would be no hacks in code, and the field would be required, which is also desirable.

**Comment by lubomir** [comment]
Looks good to me. Please squash before merging.

**Comment by xychu** [comment]
:+1:

**Pull request merged**